x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/components/tables/utils.test.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SchemaConflictFieldTypes, SchemaConflicts } from '../../../../../shared/types';
+import { EngineDetails } from '../../../engine/types';
+
+import {
+  getConflictingEnginesFromConflictingField,
+  getConflictingEnginesFromSchemaConflicts,
+  getConflictingEnginesSet,
+} from './utils';
+
+describe('getConflictingEnginesFromConflictingField', () => {
+  const CONFLICTING_FIELD: SchemaConflictFieldTypes = {
+    text: ['source-engine-1'],
+    number: ['source-engine-2', 'source-engine-3'],
+    geolocation: ['source-engine-4'],
+    date: ['source-engine-5', 'source-engine-6'],
+  };
+
+  it('returns a flat array of all engines with conflicts across different schema types, including duplicates', () => {
+    const result = getConflictingEnginesFromConflictingField(CONFLICTING_FIELD);
+
+    // we can't guaruntee ordering
```

Comment:
Optional, small typo nit
```suggestion
    // we can't guarantee ordering
```